lib/uwb/UwbDevice.cxx

```diff
@@ -41,7 +41,15 @@ UwbDevice::OnStatusChanged([[maybe_unused]] UwbStatus status)
 void
 UwbDevice::OnDeviceStatusChanged([[maybe_unused]] UwbStatusDevice statusDevice)
 {
-    // TODO: implement this
+    if (m_status.State == UwbDeviceState::Uninitialized) {
+        m_status = statusDevice;
+        PLOG_VERBOSE << "initialized device status as " << m_status.ToString();
+        return;
+    }
+    if (m_status.State != statusDevice.State) {
+        PLOG_VERBOSE << "changed state: " << m_status.ToString() << " --> " << statusDevice.ToString();
```

Comment:
nit: We have to `ToString()` helper such that you can do this instead without invoking `ToString()` directly:

```
#include <notstd/tostring.hxx>
using namespace strings::ostream_operators
```
...
```suggestion
    if (m_status.State == UwbDeviceState::Uninitialized) {
        m_status = statusDevice;
        PLOG_VERBOSE << "initialized device status as " << m_status;
        return;
    }
    if (m_status.State != statusDevice.State) {
        PLOG_VERBOSE << "changed state: " << m_status << " --> " << statusDevice;
```

Reply:
I tried your suggestion but it complained it couldn't find the operator<< for m_status :(

Reply:
Hmm, probably the `PLOG` macro isn't actually an `ostream`. Oh well :'(